internal/adapters/operation/operation_storage_redis.go

```diff
@@ -350,6 +392,41 @@ func (r *redisOperationStorage) getFinishedOperationsFromHistory(ctx context.Con
 	return operationsIDs, err
 }
 
+func (r *redisOperationStorage) updateOperationInSortedSet(ctx context.Context, pipe redis.Pipeliner, schedulerName string, op *operation.Operation, newStatus operation.Status) error {
+	var listKey string
+	operationTookAction, err := operationHasNoAction(op)
+	if err != nil {
+		return errors.NewErrUnexpected("failed to check if operation took action when updating status").WithError(err)
+	}
+
+	switch {
+	case newStatus == operation.StatusInProgress:
+		listKey = r.buildSchedulerActiveOperationsKey(schedulerName)
+	case newStatus == operation.StatusFinished && operationTookAction:
+		listKey = r.buildSchedulerNoActionKey(schedulerName)
+	default:
+		listKey = r.buildSchedulerHistoryOperationsKey(schedulerName)
+	}
+
+	pipe.ZRem(ctx, r.buildSchedulerActiveOperationsKey(schedulerName), op.ID)
+	pipe.ZRem(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName), op.ID)
+	pipe.ZRem(ctx, r.buildSchedulerNoActionKey(schedulerName), op.ID)
+	pipe.ZAdd(ctx, listKey, &redis.Z{Member: op.ID, Score: float64(r.clock.Now().Unix())})
+	return nil
+}
+
+func operationHasNoAction(op *operation.Operation) (bool, error) {
+	if op.DefinitionName == healthcontroller.OperationName {
+		def := healthcontroller.SchedulerHealthControllerDefinition{}
+		err := def.Unmarshal(op.Input)
+		if err != nil {
+			return false, err
+		}
+		return !*def.TookAction, nil
+	}
+	return false, nil
+}
```

Comment:
Hey @guilhermocc this is confusing me. Here the function search for `noop`, but the return is to a variable called `operationTookAction`.

Reply:
And during the switch, it is tested as if `operationTookAction`